src/integrationTest/java/uk/gov/hmcts/darts/retention/service/ApplyRetentionProcessorIntTest.java

```diff
@@ -0,0 +1,112 @@
+        courtCase = dartsDatabase.createCase(
+                "a courthouse",
+                "a case"
+        );
+        courtCase.setClosed(true);
+        dartsDatabase.save(courtCase);
+
+        OffsetDateTime retainUntilDate = OffsetDateTime.parse("2030-01-01T12:00Z");
+
+        CaseRetentionEntity caseRetentionEntity = dartsDatabase.createCaseRetentionObject(courtCase, CaseRetentionStatus.PENDING, retainUntilDate, false);
+        caseRetentionEntity.setCreatedDateTime(OffsetDateTime.now().minusDays(8));
+        caseRetentionRepository.saveAndFlush(caseRetentionEntity);
+
+    }
+
+    @Test
+    void testCaseRetentionChangeState() {
+        applyRetentionProcessor = new ApplyRetentionProcessorImpl(caseRetentionRepository, currentTimeHelper);
+
+        List<CaseRetentionEntity> caseRetentionEntities = caseRetentionRepository.findAllByCourtCase(courtCase);
+        assertEquals(CaseRetentionStatus.PENDING.name(), caseRetentionEntities.get(0).getCurrentState());
+        applyRetentionProcessor.processApplyRetention();
+
+        caseRetentionEntities = caseRetentionRepository.findAllByCourtCase(courtCase);
+
+        assertTrue(caseRetentionEntities.get(0).getCreatedDateTime().isBefore(OffsetDateTime.now().minusDays(7)));
+        assertEquals(caseRetentionEntities.get(0).getRetainUntilAppliedOn().truncatedTo(ChronoUnit.DAYS), OffsetDateTime.now().truncatedTo(ChronoUnit.DAYS));
+        assertEquals(CaseRetentionStatus.COMPLETE.name(), caseRetentionEntities.get(0).getCurrentState());
+    }
+
+    @Test
+    void testCaseRetentionChangeStateWithRecordInsideCoolOff() {
+        OffsetDateTime retainUntilDate = OffsetDateTime.parse("2030-01-01T12:00Z");
+
+        CaseRetentionEntity caseRetentionEntity = dartsDatabase.createCaseRetentionObject(courtCase, CaseRetentionStatus.PENDING, retainUntilDate, false);
+        caseRetentionEntity.setCreatedDateTime(OffsetDateTime.now().minusDays(6));
+        caseRetentionRepository.saveAndFlush(caseRetentionEntity);
+
+        applyRetentionProcessor = new ApplyRetentionProcessorImpl(caseRetentionRepository, currentTimeHelper);
+
+        List<CaseRetentionEntity> caseRetentionEntities = caseRetentionRepository.findAllByCourtCase(courtCase);
+        assertEquals(CaseRetentionStatus.PENDING.name(), caseRetentionEntities.get(0).getCurrentState());
+        applyRetentionProcessor.processApplyRetention();
+
+        caseRetentionEntities = caseRetentionRepository.findAllByCourtCase(courtCase);
+        assertTrue(caseRetentionEntities.get(0).getCreatedDateTime().isBefore(OffsetDateTime.now().minusDays(7)));
+        assertEquals(caseRetentionEntities.get(0).getRetainUntilAppliedOn().truncatedTo(ChronoUnit.DAYS), OffsetDateTime.now().truncatedTo(ChronoUnit.DAYS));
+        assertEquals(CaseRetentionStatus.COMPLETE.name(), caseRetentionEntities.get(0).getCurrentState());
+
+        assertEquals(CaseRetentionStatus.PENDING.name(), caseRetentionEntities.get(1).getCurrentState());
```

Comment:
caseRetentionEntities hasn't been ordered, can you confirm it will always bring them back in this order?